wrapper/src/main/java/software/amazon/jdbc/plugin/federatedauth/CredentialsProviderGetterFactory.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.federatedauth;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.PluginService;
+
+public interface CredentialsProviderGetterFactory {
```

Comment:
omg.

Reply:
i know.... :(

Reply:
Yeah....if I rename `CredentialsProviderGetter` to `CredentialsProviderFactory`, then this class becomes `CredentialsProviderFactoryFactory` which is a bit funny too. Not sure if this is maybe a code smell.

Reply:
Currently looking at the Abstract Factory pattern. I think this pattern might fit this scenario a bit better, and eliminate this funky naming.

Reply:
Just pushed change to use Abstract Factory pattern instead of a Factory of Factories. :)